File: client/src/app/modules/quotation/services/service.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { ServiceType } from '../model/ServiceType';

@Injectable({
  providedIn: 'root'
})
export class ServiceTypeService extends AppRestService<ServiceType>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getServiceTypes() {
    return this.getList(new HttpParams(), "service-types");
  }

  addOrUpdateServiceType(service: ServiceType) {
    return this.addOrUpdate(new HttpParams(), "service-type", service, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/quotation/service/ServiceTypeService.java
package com.jss.osiris.modules.quotation.service;

import java.util.List;

import com.jss.osiris.modules.quotation.model.ServiceType;

public interface ServiceTypeService {
    public List<ServiceType> getServiceTypes();

    public ServiceType getServiceType(Integer id);

    public ServiceType addOrUpdateServiceType(ServiceType service);
}
